src/cpu/aarch64/matmul/acl_matmul.hpp

```diff
@@ -92,8 +102,18 @@ struct acl_matmul_t : public primitive_t {
             VDISPATCH_MATMUL(!has_runtime_dims_or_strides(),
                     VERBOSE_RUNTIMEDIM_UNSUPPORTED);
 
-            CHECK(acl_matmul_utils::init_conf_matmul(
-                    amp_, src_md_, weights_md_, dst_md_, *desc(), *attr()));
+            if (weights_format_kind == format_kind::any) {
+                CHECK(acl_matmul_utils::init_conf_matmul_fixed_format(
+                        amp_, src_md_, weights_md_, dst_md_, *desc(), *attr()));
+            } else {
+#if DNNL_CPU_THREADING_RUNTIME == DNNL_RUNTIME_THREADPOOL
+                // to avoid seg. fault in case threadpool is enabled and its pointer is null
+                if (threadpool_utils::get_active_threadpool() == nullptr)
+                    return status::unimplemented;
+#endif
+                CHECK(acl_matmul_utils::init_conf_matmul_non_fixed_format(
+                        amp_, src_md_, weights_md_, dst_md_, *desc(), *attr()));
+            }
```

Comment:
Minor: does the naming really go like that? I thought that "non-fixed" is referred to "format_kind::any", not the other way around...

Reply:
The name is correct. The fixed format means that once a kernel and therefore format has been chosen, it does not change. Hence, it is a fixed-format kernel. "format_kind::any" indicates to the ACL library that it can choose the format.